train.py

```diff
@@ -73,12 +75,23 @@ def generate_subspectrogram(duration_s=config.SUBSPECTROGRAM_DURATION_S, fft_rat
         checkpoint.epoch.assign_add(1)
         checkpoint_manager.save()
 
-    print("======================== evaluation on validation data =========================")
-    # test model on validation set
-    utils.test_model(model, val_spectrograms, val_labels, val_loss, val_accuracy)
+        if epoch % 10 == 0 or epoch == config.NB_EPOCHS - 1:
+            print("======================== evaluation on validation data =========================")
+            # test model on validation set
+            this_val_accuracy, this_val_loss = utils.test_model(
+                model, val_spectrograms, val_labels, val_loss, val_accuracy, epoch=epoch+1)
+            val_accuracies.append(this_val_accuracy)
+            val_losses.append(this_val_loss)
 
     utils.save_model(model, epoch)
-    utils.save_and_display_loss_through_epochs(epoch_range, loss_history, model.name)
+    utils.save_and_display_loss_through_epochs(epoch_range, loss_history, model.name, "training loss")
+
+    # Plot val accuracies
+    val_range = [i for i in epoch_range if i % 10 == 0 or i == config.NB_EPOCHS - 1]
+    utils.save_and_display_loss_through_epochs(val_range, val_accuracies, model.name, "validation accuracy")
+    utils.save_and_display_loss_through_epochs(val_range, val_losses, model.name, "validation loss")
+    best_epoch = (len(val_accuracies) - list(reversed(val_accuracies)).index(min(val_accuracies)) - 1) * 10
```

Comment:
Why not just `val_loss.index(min(val_loss)) * 10` ?

Reply:
Because I want the last one, not the first.